tck/src/main/java/org/eclipse/microprofile/config/tck/ArrayConverterTest.java

```diff
@@ -1654,120 +1658,161 @@ public void testInstantSetInjection() {
 
     ///////////////////////////////////Test URL[] //////////////////////////
 
+    private void assertURLArrayEquals(URL[] value) throws MalformedURLException {
+        URL[] expectedValue = new URL[]{
+            new URL("http://microprofile.io"),
+            new URL("http://openliberty.io"),
+            new URL("http://microprofile.io")};
+        Assert.assertTrue(IntStream.range(0, expectedValue.length).allMatch(i -> {
+        try {
+                return expectedValue[i].toURI().equals(value[i].toURI());
+        } 
+        catch (URISyntaxException e) {
+                e.printStackTrace();
+        }
+        return false;
+        }));
+    }
+
+    private void assertURLListEquals(List<URL> value) throws MalformedURLException {
+        List<URL> expectedValue = Arrays.asList(new URL[]{
+            new URL("http://microprofile.io"),
+            new URL("http://openliberty.io"),
+            new URL("http://microprofile.io")});
+        Assert.assertTrue(IntStream.range(0, expectedValue.size()).allMatch(i -> {
+        try {
+                return expectedValue.get(i).toURI().equals(value.get(i).toURI());
+        } 
+        catch (URISyntaxException e) {
+                e.printStackTrace();
+        }
+        return false;
+        }));
+    }
+
+
     @Test
-    public void testUrlArrayLookupProgrammatically() throws MalformedURLException {
+    public void testUrlArrayLookupProgrammatically() throws MalformedURLException, URISyntaxException {
         URL[] value = config.getValue("tck.config.test.javaconfig.converter.urlvalues", URL[].class);
         Assert.assertNotNull(value);
         Assert.assertEquals(value.length, 3);
```

Comment:
If there is a special method to assert equality I'd think the length and null check can or should be part of it so they do not need duplication.

Reply:
I think this is a short circuit. The whole test will be failed if the value is null or contains a different size of the items.